osu.Game/Online/Multiplayer/MultiplayerClient.cs

```diff
@@ -32,7 +32,7 @@ public abstract class MultiplayerClient : Component, IMultiplayerClient, IMultip
         /// <summary>
         /// Invoked when any change occurs to the multiplayer room.
         /// </summary>
-        public event Action? RoomUpdated;
+        public virtual event Action? RoomUpdated;
```

Comment:
Not sure how to feel about adding these virtual specs. I had approved this kind of code before with the thought process being that this is going to be done sparingly, but I much prefer interface mocking and therefore I'd kind of loath to see this become the norm (which it seems like it might because https://github.com/ppy/osu/pull/17767 did this too). Maybe fine now that there's precedent...?

Reply:
I agree. It should be easy enough to switch if/when we choose. These are detected as unused by rider and can be removed in one sweep.